src/main/scala/xiangshan/cache/mmu/L2TLB.scala

```diff
@@ -387,13 +444,64 @@ class L2TLBImp(outer: L2TLB)(implicit p: Parameters) extends PtwModule(outer) wi
   // (miss queue may can't resp to tlb with low latency, it should have highest priority, but diffcult to design cache)
-  val resp_pte_sector = VecInit((0 until MemReqWidth).map(i =>
+  val resp_pte_sector = VecInit((0 until (if(HasBitmapCheck) MemReqWidth/2 else MemReqWidth)).map(i =>
     if (i == l2tlbParams.llptwsize + 1) {RegEnable(refill_data_tmp, 0.U.asTypeOf(refill_data_tmp), mem_resp_done && mem_resp_from_hptw) }
     else if (i == l2tlbParams.llptwsize) {RegEnable(refill_data_tmp, 0.U.asTypeOf(refill_data_tmp), mem_resp_done && mem_resp_from_ptw) }
     else { Mux(llptw_mem.buffer_it(i), refill_data, RegEnable(refill_data, 0.U.asTypeOf(refill_data), llptw_mem.buffer_it(i))) }
     // llptw could not use refill_data_tmp, because enq bypass's result works at next cycle
   ))
 
+  if(HasBitmapCheck){
+    //add bitmap arb
+    bitmap.foreach { Bitmap =>
+    val bitmap_arb = Module(new Arbiter(new bitmapReqBundle(), 3))
+    bitmap_arb.io.in(0) <> ptw.io.bitmap.req
+    bitmap_arb.io.in(1) <> llptw.io.bitmap.req
+    bitmap_arb.io.in(2) <> hptw.io.bitmap.req
+    bitmap_arb.io.out.ready := Bitmap.io.req.ready
+
+    Bitmap.io.req <> bitmap_arb.io.out
+
+    //connect bitmap resp to PTW
+    val bitmapresp_to_llptw = from_llptw(Bitmap.io.resp.bits.id)
+    val bitmapresp_to_hptw = from_hptw(Bitmap.io.resp.bits.id)
+    val bitmapresp_to_ptw = from_ptw(Bitmap.io.resp.bits.id)
+
+    Bitmap.io.resp.ready := (llptw.io.bitmap.resp.ready && bitmapresp_to_llptw) || (hptw.io.bitmap.resp.ready && bitmapresp_to_hptw) || (ptw.io.bitmap.resp.ready && bitmapresp_to_ptw)
+
+    //bitmap -> llptw ptw hptw
+    llptw.io.bitmap.resp.valid := Bitmap.io.resp.valid && bitmapresp_to_llptw
+    hptw.io.bitmap.resp.valid := Bitmap.io.resp.valid && bitmapresp_to_hptw
+    ptw.io.bitmap.resp.valid := Bitmap.io.resp.valid && bitmapresp_to_ptw
+
+    //add ptw、hptw、llptw with bitmap resp connect
+    ptw.io.bitmap.resp.bits := Bitmap.io.resp.bits
+    hptw.io.bitmap.resp.bits := Bitmap.io.resp.bits
+    llptw.io.bitmap.resp.bits := Bitmap.io.resp.bits
+
+    //mem -> bitmap
+    Bitmap.io.mem.resp.valid := mem_resp_done && mem_resp_from_bitmap
+    Bitmap.io.mem.resp.bits.id := DataHoldBypass(mem.d.bits.source, mem.d.valid)
+    Bitmap.io.mem.resp.bits.value := DataHoldBypass(refill_data_tmp.asUInt, mem.d.valid)
+    }
+  }else{
+    llptw.io.bitmap.resp.valid := DontCare
+    hptw.io.bitmap.resp.valid := DontCare
+    ptw.io.bitmap.resp.valid := DontCare
+    ptw.io.bitmap.resp.bits := DontCare
+    hptw.io.bitmap.resp.bits := DontCare
+    llptw.io.bitmap.resp.bits := DontCare
+    ptw.io.bitmap.req.ready := DontCare
+    hptw.io.bitmap.req.ready := DontCare
+    llptw.io.bitmap.req.ready := DontCare
+    ptw.io.bitmap.req.bits := DontCare
+    hptw.io.bitmap.req.bits := DontCare
+    llptw.io.bitmap.req.bits := DontCare
+  }
+  //ptwcache -> hptw llptw
+  hptw.io.l0_way_info := cache.io.l0_way_info
+  llptw.io.l0_way_info := cache.io.l0_way_info
+
```

Comment:
1. Please put a space after // in the comment.
2. `if(xxx){` -> `if (xxx) {`
3. `}else{` -> `} else {`
4. These bundles should be set as option instead of waiting for chisel auto-optimisation with DontCare in non-bitmap scenarios